### ISSUE TITLE
Detect enum and option values

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#93]
Example:

```
typedef NS_ENUM(NSUInteger, XUInfoPageFormat) {
	XUInfoPageFormatWebPage,
	XUInfoPageFormatXML,
	XUInfoPageFormatJSON
};

... = XUInfoPageFormatXML;
```

If there is a symbol that's stand-alone, see if it has a prefix of some enum - if yes, convert it to the shortened syntax:

```
... = .XML
```

==========================================

With options, this is pretty much the same:

```
typedef NS_OPTIONS(NSUInteger, XURegexOptions) {
	XURegexOptionNone = 0,
	XURegexOptionCaseless = 1 << 0,
	XURegexOptionMultiline = 1 << 1,
	XURegexOptionNotGreedy = 1 << 2
};

... = XURegexOptionCaseless
```

Except that with options you are likely to have the options name ending with an extras `s`: XURegexOption**s**

```
... = .Caseless
```

**Comment by akac** [comment]
This is my #1 desire. Nobody does this yet.

**Comment by drkameleon** [comment]
OK,... so here are some good news for you. That was *very* tricky, but I believe I made it! :)

So... closing this as fixed, as of the upcoming 2.1. Given that this one *is* tricky and very experimental, once 2.1 is released (it'll be very soon), if you have any issue, please feel free to let me know and have the issue re-opened.

Re: This fix comes together with another interesting fix (caught it yesterday myself...) - https://github.com/drkameleon/iswift/issues/134

---

As per the initial example....

**Example:**
```objective-c
typedef NS_ENUM(NSUInteger, XUInfoPageFormat) {
    XUInfoPageFormatWebPage,
    XUInfoPageFormatXML,
    XUInfoPageFormatJSON
};

typedef NS_OPTIONS(NSUInteger, XURegexOptions) {
    XURegexOptionNone = 0,
    XURegexOptionCaseless = 1 << 0,
    XURegexOptionMultiline = 1 << 1,
    XURegexOptionNotGreedy = 1 << 2
};

@implementation aClass

- (void)aFunc {
    NSUInteger a = XUInfoPageFormatXML;
    NSUInteger b = XURegexOptionCaseless;
}
    
@end
```

**Output:**
```swift
enum XUInfoPageFormat {
	case XUInfoPageFormatWebPage, XUInfoPageFormatXML, XUInfoPageFormatJSON
}
enum XURegexOptions {
	case XURegexOptionNone, XURegexOptionCaseless, XURegexOptionMultiline, XURegexOptionNotGreedy
}

class aClass {

	func aFunc() {
		var a: UInt = .XML
		var b: UInt = .Caseless
	}
	
}
```
